pintos/src/userprog/syscall.c

```diff
@@ -68,7 +118,75 @@ syscall_handler (struct intr_frame *f)
   }
 }
 
-oid sys_halt(void) {
+/* Opens the file called file. Returns a nonnegative integer handle called a "file descriptor" or -1 if the file could
+ * not be opened. The file descriptor will be the integer location of the file in the current thread's list of files
+ * */
+int sys_open(char * file)
+{
+  // obtain lock for filesystem since we are about to open the file
+  lock_acquire(&filesys_lock);
+
+  // open the file
+  struct file * new_file_struct = filesys_open(file);
+
+  // all done with file sys for now
+  lock_release(&filesys_lock);
+
+  // file will be null if file not found in file system
+  if (file==NULL){
+    // nothing to do here open fails, return -1
+    return -1;
+  }
+  // else add file to current threads list of open files
+  // from pintos notes section 3.3.4 System calls: when a single file is opened more than once, whether by a single
+  // process or different processes each open returns a new file descriptor. Different file descriptors for a single
+  // file are closed independently in seperate calls to close and they do not share a file position. We should make a
+  // list of files so if a single file is opened more than once we can close it without conflicts.
+  struct thread_file * new_thread_file = malloc(sizeof(struct thread_file));
+  new_thread_file->file_struct = new_file_struct;
+  new_thread_file->fd = thread_current()->next_fd;
+  thread_current()->next_fd++;
+  list_push_back(&thread_current()->open_files, &new_thread_file->list_elem_struct);
+  return new_thread_file->fd;
+}
+
+int sys_exec (const char *cmdline){
```

Comment:
nice catch, pid_t is type defined to be int in user/syscall.h, but it would be safer to have the return type be pid_t here thanks, ill change that now